File: Languages/Zig/rand_avg.zig
const std = @import("std");
const fs = std.fs;
const print = std.debug.print;

// Hello, World!
fn hello() void {
    print("Hello, World!\n");
}

fn generateRandomNumbers() !void {
    const count = 1000;
    const output_dir = "../rand_avg output";
    const output_file = "../rand_avg output/random_numbers.txt";
    
    // Create output directory
    try fs.cwd().makePath(output_dir);
    
    // Random number generator
    var prng = std.rand.DefaultPrng.init(@intCast(std.time.timestamp()));
    var random = prng.random();
    
    // Generate 1000 random numbers
    var sum: u64 = 0;
    
    const file = try fs.cwd().createFile(output_file, .{});
    defer file.close();
    
    var writer = file.writer();
    
    for (0..count) |_| {
        const num = random.intRangeAtMost(u32, 0, 999);
        sum += num;
        try writer.print("{}\n", .{num});
    }
    
    // Calculate mean
    const mean = @as(f64, @floatFromInt(sum)) / @as(f64, @floatFromInt(count));
    
    print("Generated 1000 random numbers\n");
    print("Mean: {d:.2}\n", .{mean});
    print("Saved to: {s}\n", .{output_file});
}

pub fn main() void {
    hello();
    
    if (generateRandomNumbers()) |err| {
        print("Error: {}\n", .{err});
    } else |_| {
        // Success
    }
}

File: Languages/Go/rand_avg/rand_avg.go
package main

import (
	"fmt"
	"math/rand"
	"os"
	"time"
)

// Hello, World!
func hello() {
	fmt.Println("Hello, World!")
}

func generateRandomNumbers() {
	const count = 1000
	const outputDir = "../../rand_avg output"
	const outputFile = outputDir + "/random_numbers.txt"

	// Create output directory
	if err := os.MkdirAll(outputDir, 0755); err != nil {
		fmt.Println("Could not create output directory")
		return
	}

	// Seed random number generator
	rand.Seed(time.Now().UnixNano())

	// Generate 1000 random numbers
	randomNumbers := make([]int, count)
	sum := 0

	for i := 0; i < count; i++ {
		num := rand.Intn(1000)
		randomNumbers[i] = num
		sum += num
	}

	// Calculate mean
	mean := float64(sum) / float64(count)

	// Save to file
	file, err := os.Create(outputFile)
	if err != nil {
		fmt.Println("Could not create output file")
		return
	}
	defer file.Close()

	for _, num := range randomNumbers {
		_, err := file.WriteString(fmt.Sprintf("%d\n", num))
		if err != nil {
			fmt.Println("Error writing to file")
			return
		}
	}

	fmt.Println("Generated 1000 random numbers")
	fmt.Printf("Mean: %.2f\n", mean)
	fmt.Printf("Saved to: %s\n", outputFile)
}

func main() {
	hello()
	generateRandomNumbers()
}
